llvm/test/Feature/Repo/repo_pruning_dependents.ll

```diff
@@ -1,30 +1,51 @@
-; This testcase tests the program repository pruning pass.
+; This test checks the program repository pruning pass for the dependents.
+;
+; In the llvm-project-prepo project, the GO's digest could be generated either by the
+; RepoMetadataGenerator pass or by the RepoObjectWriter. When a new global object (GO)
+; is generated by an LLVM optimisation pass after the RepoMetadataGenerator pass, the
+; digest will be generated by the RepoObjectWriter. If a fragment has an external fixup
+; referencing this GO, we add the GO's compilation member address to the dependents.
+;
+; When a fragment is pruned, all its dependents need to be recorded in the repo.tickets
+; metadata to guarantee that the dependents (those GO generated by later optimizations)
+; are present in the final compilation record.
+;
+; In the test, the digest of 'A' is given in the TicketNode (as if it were generated
+; by the RepoMetadataGenerator pass). The digests of `B` and `C` will be generated by the
+; RepoObjectWriter. Since Function 'A' is dependent on function 'B' and 'B' is dependent
+; on function 'C', both TicketNodes of 'B' and 'C' need to be added into the 'repo.tickets'
+; during the pruning.
+;
+; The steps in this test are:
+; Step 1: Remove the database.
+; Step 2: Generate an IR code targeted on Repo.
+; Step 3: Remove the TicketNode metadata from the functions B and C.
+; Step 4: Since the functions B and C don't have the TicketNode, the digest of functions `B`
+;         and `C` are generated by the RepoObjectWriter and added these two functions into
+;         the dependents of the function 'A'.
```

Comment:
Throughout this text there’s a mixture of quoting styles around A, B, and C: sometimes no quotes, sometimes back-tickets, and sometimes single quotes. It would be good to stick to one (probably no quotes since they’re capitalised and already stand out a fair bit)!

Reply:
Fixed, thanks.